images/capi/ansible/roles/nvidia/tasks/main.yml

```
@@ -0,0 +1,119 @@
+# Copyright 2023 The Kubernetes Authors.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+
+- name: unload nouveau
+  modprobe:
+    name: nouveau
+    state: absent
+  ignore_errors: true
+
+- name: Add NVIDIA package signing key
+  ansible.builtin.apt_key:
+    url: https://nvidia.github.io/libnvidia-container/gpgkey
+  when: ansible_distribution == "Ubuntu"
```

```suggestion
  when: ansible_distribution == "Debian"
```

Should this be `Debian` instead? We generally prefer this if it's not specific just to Ubuntu, because it covers both distros. (Although technically I don't think we build actual Debian anywhere in the project, it seems a better default.)

Same comment for the other references to `Ubuntu` added in this file.

---

I've updated theses and spotted one more place it should be set too.